diff --git a/aports/cross/gcc4-armhf/APKBUILD b/aports/cross/gcc4-armhf/APKBUILD
index 8cdc1a7..3d04597 100644
--- a/aports/cross/gcc4-armhf/APKBUILD
+++ b/aports/cross/gcc4-armhf/APKBUILD
@@ -38,16 +38,9 @@ package() {
 
     # Unmodified package function from the gcc APKBUILD
     _package
-
-    # Workaround for: postmarketOS/binary-package-repo#1
-    echo "Replacing hardlinks with symlinks"
-    rm -v "$pkgdir"/usr/bin/"$CTARGET"-c++
-    ln -s -v /usr/bin/"$CTARGET"-g++ "$pkgdir"/usr/bin/"$CTARGET"-c++
-    rm -v "$pkgdir"/usr/bin/"$CTARGET"-gcc-"$pkgver"
-    ln -s -v /usr/bin/"$CTARGET"-gcc "$pkgdir"/usr/bin/"$CTARGET"-gcc-"$pkgver"
 }
 
-pkgname="gcc-armhf"
+pkgname="gcc4-armhf"
 pkgver=4.9.2
 _cross=""
 [ "$BOOTSTRAP" = "noheaders" ] && pkgname="gcc-pass1"
@@ -55,14 +48,14 @@ _cross=""
 [ "$CHOST" != "$CTARGET" ] && [ -n "$CHOST" -a -n "$CTARGET" ] \
 	&& _cross="-$CTARGET"
 
-pkgname="gcc-armhf"
+pkgname="gcc4-armhf"
 pkgrel=6
 pkgdesc="Stage2 cross-compiler for armhf"
 url="http://gcc.gnu.org"
 arch="aarch64 x86_64 x86"
 license="GPL LGPL"
 _gccrel=$pkgver-r$pkgrel
-depends="isl binutils-armhf"
+depends="isl binutils-armhf gcc-armhf"
 makedepends_build="gcc g++ paxmark bison flex texinfo gawk zip gmp-dev mpfr-dev mpc1-dev zlib-dev"
 makedepends_host="linux-headers gmp-dev mpfr-dev mpc1-dev isl-dev zlib-dev musl-dev-armhf binutils-armhf"
 makedepends="$makedepends_build $makedepends_host"
@@ -135,6 +128,8 @@ else
 	STRIP_FOR_TARGET=${CROSS_COMPILE}strip
 	_builddir="$srcdir/build"
 fi
+export CFLAGS="$CFLAGS -fPIC"
+export CXXFLAGS="$CXXFLAGS -fPIC"
 
 # Go needs {set,make,swap}context, unimplemented in musl
 [ "$CTARGET_LIBC" = musl ] && LANG_GO=false
@@ -357,6 +352,7 @@ build() {
 		--build=${CBUILD} \
 		--host=${CHOST} \
 		--target=${CTARGET} \
+		--program-prefix="${CTARGET}-${pkgver}-" \
 		--with-pkgversion="Alpine ${pkgver}" \
 		--enable-checking=release \
 		--disable-fixed-point \
@@ -427,10 +423,6 @@ _package() {
 	rm -f "$pkgdir"/usr/lib/libffi* "$pkgdir"/usr/share/man/man3/ffi*
 	find "$pkgdir" -name 'ffi*.h' | xargs rm -f
 
-	mkdir -p "$pkgdir"/usr/share/gdb/python/auto-load/usr/lib
-	mv "$pkgdir"/usr/lib/*-gdb.py \
-		"$pkgdir"/usr/share/gdb/python/auto-load/usr/lib/
-
 	paxmark -pmrs "$pkgdir"/$_gcclibexec/cc1 || return 1
 
 	# move ada runtime libs
@@ -446,6 +438,8 @@ _package() {
 		# native gcc package
 		rm -rf "$pkgdir"/usr/bin/cc "$pkgdir"/usr/include "$pkgdir"/usr/share
 	fi
+
+	rm "$pkgdir"/usr/$CTARGET/lib/libgcc_s.so*
 }
 
 libatomic() {
