diff --git a/arch/arm/mach-msm/board-mahimahi.c b/arch/arm/mach-msm/board-mahimahi.c
index 4f0cbbf..e398310 100644
--- a/arch/arm/mach-msm/board-mahimahi.c
+++ b/arch/arm/mach-msm/board-mahimahi.c
@@ -742,7 +742,7 @@ static uint32_t flashlight_gpio_table_rev_CX[] = {
 };
 
 
-static int config_mahimahi_flashlight_gpios(void)
+static void config_mahimahi_flashlight_gpios(void)
 {
 	if (is_cdma_version(system_rev)) {
 		config_gpio_table(flashlight_gpio_table_rev_CX,
@@ -751,7 +751,6 @@ static int config_mahimahi_flashlight_gpios(void)
 		config_gpio_table(flashlight_gpio_table,
 			ARRAY_SIZE(flashlight_gpio_table));
 	}
-	return 0;
 }
 
 static struct flashlight_platform_data mahimahi_flashlight_data = {
diff --git a/arch/arm/mach-msm/footswitch.h b/arch/arm/mach-msm/footswitch.h
index 3d9b479..cea97b2 100644
--- a/arch/arm/mach-msm/footswitch.h
+++ b/arch/arm/mach-msm/footswitch.h
@@ -54,10 +54,12 @@
 				.valid_ops_mask   = REGULATOR_CHANGE_STATUS, \
 			}, \
 			.num_consumer_supplies = 1, \
+			.regulator_init = 0, \
 			.consumer_supplies = \
 				&(struct regulator_consumer_supply) \
 				REGULATOR_SUPPLY((_name), NULL), \
 		} \
+		.power = 0 \
 	}, \
 })
 #define FS_PCOM(_id, _name) FS_GENERIC("footswitch-pcom", (_id), (_name))
diff --git a/arch/arm/mach-msm/pm.c b/arch/arm/mach-msm/pm.c
index f182233..eb3d944 100644
--- a/arch/arm/mach-msm/pm.c
+++ b/arch/arm/mach-msm/pm.c
@@ -165,12 +165,14 @@ static struct msm_pm_time_stats {
 	int count;
 	int64_t total_time;
 } msm_pm_stats[MSM_PM_STAT_COUNT] = {
-	[MSM_PM_STAT_REQUESTED_IDLE].name = "idle-request",
-	[MSM_PM_STAT_IDLE_SPIN].name = "idle-spin",
-	[MSM_PM_STAT_IDLE_WFI].name = "idle-wfi",
-	[MSM_PM_STAT_IDLE_SLEEP].name = "idle-sleep",
-	[MSM_PM_STAT_IDLE_FAILED_SLEEP].name = "idle-failed-sleep",
-	[MSM_PM_STAT_NOT_IDLE].name = "not-idle",
+#define ENT(n) ( { .name = n, .bucket = 0, .min_time = 0, .max_time=0, count=0, .total_time = 0 } )
+	[MSM_PM_STAT_REQUESTED_IDLE] = ENT("idle-request"),
+	[MSM_PM_STAT_IDLE_SPIN] = ENT("idle-spin"),
+	[MSM_PM_STAT_IDLE_WFI] = ENT("idle-wfi"),
+	[MSM_PM_STAT_IDLE_SLEEP] = ENT("idle-sleep"),
+	[MSM_PM_STAT_IDLE_FAILED_SLEEP] = ENT("idle-failed-sleep"),
+	[MSM_PM_STAT_NOT_IDLE] = ENT("not-idle"),
+#undef ENT
 };
 
 static void msm_pm_add_stat(enum msm_pm_time_stats_id id, int64_t t)
@@ -682,7 +684,7 @@ void msm_pm_flush_console(void)
 	console_unlock();
 }
 
-static void msm_pm_restart(char str)
+static void msm_pm_restart(char str, const char *cmd)
 {
 	msm_pm_flush_console();
 
